src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,496 @@
+{
+  size_t idx = 0;
+  uint16_t crc;
+  uint16_t rec_buf_len = 0;
+  int command_length;
+  int rec_bufsize = 127;
+  uint8_t command[rec_bufsize];
+
+  /** loop as long we have data */
+  while (idx < len)
+  {
+    /** check if we received a header*/
+    if (data[idx] == 0xA5)
+      /** reset buffer length */
+      rec_buf_len = 0;
+
+    /** check if the buffer-size is reached */
+    if (rec_buf_len >= rec_bufsize)
+      /** reset buffer length */
+      rec_buf_len = 0;
+
+    /** write command in buffer */
+    command[rec_buf_len++] = (char)data[idx++];
+
+    /** if we received data we have to find the start of the protocol
+     *  for this we truncate the message until we find the start
+     *  of the protocol which is 0x14 (PROT-ID). For this we have
+     * a look at the first two bytes of the command.
+     * If we find this id we go on and try to parse the data below.
+    */
+    if (rec_buf_len == 2)
+    {
+      /** 
+       * check if command is NOT PROT-ID 
+       * 0x14 is the start of the PROT
+       */
+      if (!(command[rec_buf_len - 1] == 0x14))
+      {
+        /** reset buffer length */
+        rec_buf_len = 0;
+      }
+    }
+
+    /** we got some data and the data begins with the PROT-ID */
+    else if (rec_buf_len >= 3)
+    {
+      command_length = command[2];
+
+      /** all received */
+      if (rec_buf_len >= (command_length + 5))
+      {
+        /** generate uint16_t out of commands */
+        uint16_t crc_value = BytesToUnsigned(command[command_length + 4], command[command_length + 3]);
+
+        /** generate crc */
+        crc = CRCBitwise(command, command_length + 3);
+
+        /** check crc */
+        if (crc == crc_value || crc_value == 0)
+        {
+          /** extract and save active and passive frequency */
+          SaveAR62xxFrequencies(command, command_length + 5, crc);
+        }
+        /** reset buffer length */
+        rec_buf_len = 0;
+      }
+    }
+  }
+  return true;
+}
+
+void AR62xxDevice::SaveAR62xxFrequencies(uint8_t *sz_command, unsigned int len, uint16_t crc)
+{
+  /** command 22 = frequency-settings
+   *  Frequency settings, always for both frequencies (active and passive) */
+  if ((unsigned char)(sz_command[3] & 0x7F) == 22)
+  {
+    active_frequency = AR62xxIdToFrequency(BytesToUnsigned(sz_command[5], sz_command[4]));
+    passive_frequency = AR62xxIdToFrequency(BytesToUnsigned(sz_command[7], sz_command[6]));
+  }
+}
+
+bool AR62xxDevice::PutActiveFrequency(RadioFrequency frequency,
+                                      const TCHAR *name,
+                                      OperationEnvironment &env)
+{
+  /** command */
+  uint8_t sz_tmp[MAX_CMD_LEN];
+
+  /** generate the command to set the active station */
+  unsigned int len = GenerateSetStationCommand(sz_tmp, ACTIVE_STATION, frequency);
+
+  /** send the command */
+  bool isSend = Send((uint8_t *)&sz_tmp, len, env);
+
+  /** return if command could be send */
+  return isSend;
```

Comment:
Was `return Send(...)` too easy for you?

Reply:
replaced